### PR TITLE
Added Cell execution timestamp (issue #68)

### PR DESCRIPTION
**Pull request by indig**
Also:
* error to stderr if there's a problem writing the notebook
* lots of CSS tweaks that are hopefully not too wacky
* repr UI now hides tabs if there's only one of 'em

### CODE REVIEW

File: polynote-frontend/polynote/cell.js [hunk 1]
```diff
@@ -378,49 +385,69 @@ export class CodeCell extends Cell {
...
                 });
+
             }
         } else {
             throw "Result must be a ResultValue"
         }
     }
 
+    setExecutionInfo(result) {
+        console.log("got exec info!", result, result.timestamp, result.durationMs, Number(result.timestamp), new Date(Number(result.timestamp)));
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/cell.js]: console.log

File: polynote-frontend/polynote/codec.js [hunk 2]
```diff
@@ -50,9 +50,9 @@ export class DataReader {
 
     // NOTE: JavaScript can't represent all 64-bit integers...
     readInt64() {
-        const hi = this.readInt32();
-        const lo = this.readInt32();
-        return (hi << 32) | lo;
+        const result = this.buffer.getBigInt64(this.offset);
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/codec.js]: 😲 is this a Chrome-specific thing?

File: polynote-frontend/polynote/cell.js (hunk 1 above)
Comment (indig) [reply polynote-frontend/polynote/cell.js]: Oof I thought I caught all of them! Thanks

File: polynote-frontend/polynote/codec.js (hunk 2 above)
Comment (indig) [reply polynote-frontend/polynote/codec.js]: Chrome, FF, Opera looks like: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#Browser_compatibility

File: polynote-frontend/polynote/cell.js (hunk 1 above)
Comment (indig) [reply polynote-frontend/polynote/cell.js]: Fixed!

File: polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala [hunk 3]
```diff
@@ -173,6 +176,16 @@ class PolyKernel private[kernel] (
                       IO.pure(result)
                   }
                 }
+
+                for {
+                  _ <- symbolTable.drain()
+                  start <- clock.monotonic(MILLISECONDS)
```
Comment (jeremyrsmith) [inline polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala]: This is neat! Didn't even know this existed!

File: polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala [hunk 4]
```diff
@@ -173,6 +176,16 @@ class PolyKernel private[kernel] (
...
+                  res <- results
+                  end <- clock.monotonic(MILLISECONDS)
+                  timestamp <- clock.realTime(MILLISECONDS)
```
Comment (jeremyrsmith) [inline polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala]: Should the timestamp be when the execution *started*? TBH I'm not sure what people would expect (or if they care one way or the other)

File: polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala (hunk 3 above)
Comment (indig) [reply polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala]: Yeah it's pretty cool :) 
(once I figured out how to make one lol - `Clock.create` is not mentioned at all in the docs!).

File: polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala (hunk 4 above)
Comment (indig) [reply polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala]: That's a good question. I just tried on Zep and it looks like the end time is used there.

File: polynote-frontend/polynote/cell.js [hunk 5]
```diff
@@ -378,49 +385,68 @@ export class CodeCell extends Cell {
...
+                const outLabel = div(['out-ident', 'with-reprs'], `Out:`);
+                this.resultTabs.appendChild(outLabel);
 
-                this.addOutput(mime, content).then(el => {
-                    el.reprs = result.reprs;
-                    this.resultEl = el;
-                    el.insertBefore(outLabel, el.childNodes[0])
+                const [mime, content] = result.displayRepr;
+                const self = this;
+                this.buildOutput(mime, content).then(function(el) {
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/cell.js]: Note: if you use the arrow lambda syntax `el => {...}` then you get to keep `this` and you don't have to rebind it to `self`.

File: polynote-frontend/polynote/cell.js [hunk 6]
```diff
@@ -378,49 +385,68 @@ export class CodeCell extends Cell {
...
+                    self.resultTabs.appendChild(el);
+                    const reprUi = new ReprUI(`Cell${self.id}`, self.path, result.reprs, el);
+                    reprUi.setEventParent(self);
+                    reprUi.show();
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/cell.js]: I think we need to make the default tab show here.. currently it goes straight to the Plot tab (for a DataFrame) which is kind of unhelpful.

File: polynote-frontend/polynote/cell.js (hunk 5 above)
Comment (indig) [reply polynote-frontend/polynote/cell.js]: Good to know - thanks!

**Pull request merged**